src/Aspire.Hosting/ApplicationModel/ResourceUrlAnnotation.cs

```diff
@@ -30,4 +30,24 @@ public sealed class ResourceUrlAnnotation : IResourceAnnotation
     /// The display order the URL. Higher values mean sort higher in the list.
     /// </summary>
     public int? DisplayOrder;
+
+    /// <summary>
+    /// Whether this URL should be shown on the dashboard resources page. Defaults to <see langword="true"/>.
+    /// </summary>
+    /// <remarks>
+    /// Set this to <see langword="false"/> to only show this URL in the resource details grid.
+    /// </remarks>
+    public bool ShowOnResourcesPage { get; set; } = true;
+
+    internal ResourceUrlAnnotation WithEndpoint(EndpointReference endpoint)
+    {
+        return new()
+        {
+            Url = Url,
+            DisplayText = DisplayText,
+            Endpoint = endpoint,
+            DisplayOrder = DisplayOrder,
+            ShowOnResourcesPage = ShowOnResourcesPage
+        };
+    }
```

Comment:
This was added to support the new overload of `WithUrlForEndpoint` that allows adding a new URL for an endpoint. The callback in that case has the user return the `ResourceUrlAnnotation` and then we call this method to ensure the correct endpoint is associated with it. The user could set it themselves in the callback as we pass them the `EndpointReference`, but this way ensures it's always correct. The `Endpoint` property is currently marked `init` and it didn't seem to me that anything else warranted changing it to `set` but I could be convinced otherwise.